unsupported/Eigen/CXX11/src/Tensor/TensorReduction.h

```diff
@@ -774,14 +778,22 @@ struct TensorEvaluator<const TensorReductionOp<Op, Dims, ArgType, MakePointer_>,
   // Indexed by reduced dimensions.
   array<Index, NumReducedDims> m_reducedDims;
 
+#if defined(EIGEN_HIPCC)
+ public:
```

Comment:
@deven-amd please help check if this can be removed. I recall this was necessary back in `hipeigen` project to workaround some limitations within `hcc` but not sure if it's still needed now